examples/Cassie/cassie_utils.cc

```diff
@@ -100,30 +100,32 @@ void buildCassieTree(RigidBodyTree<double>& tree, std::string filename,
                              rod_on_thigh_right, achilles_length);
 
   // Add spring forces
-  int body_index = tree.FindIndexOfChildBodyOfJoint("knee_joint_left");
-  auto body = tree.get_mutable_body(body_index);
-  RevoluteJoint& knee_joint_left =
-      dynamic_cast<RevoluteJoint&>(body->get_mutable_joint());
-  // stiffness is 2300 in URDF,these #s from gazebo
-  knee_joint_left.SetSpringDynamics(1500.0, 0.0);
-
-  body_index = tree.FindIndexOfChildBodyOfJoint("knee_joint_right");
-  body = tree.get_mutable_body(body_index);
-  RevoluteJoint& knee_joint_right =
-      dynamic_cast<RevoluteJoint&>(body->get_mutable_joint());
-  knee_joint_right.SetSpringDynamics(1500.0, 0.0);  // 2300 in URDF
-
-  body_index = tree.FindIndexOfChildBodyOfJoint("ankle_spring_joint_left");
-  body = tree.get_mutable_body(body_index);
-  RevoluteJoint& ankle_spring_joint_left =
-      dynamic_cast<RevoluteJoint&>(body->get_mutable_joint());
-  ankle_spring_joint_left.SetSpringDynamics(1250.0, 0.0);  // 2000 in URDF
-
-  body_index = tree.FindIndexOfChildBodyOfJoint("ankle_spring_joint_right");
-  body = tree.get_mutable_body(body_index);
-  RevoluteJoint& ankle_spring_joint_right =
-      dynamic_cast<RevoluteJoint&>(body->get_mutable_joint());
-  ankle_spring_joint_right.SetSpringDynamics(1250.0, 0.0);  // 2300 in URDF
+  if(is_with_springs){
```

Comment:
I'm not very sure how `Weld` works. Does it means that two body cannot move relatively anymore? If it is what I think, maybe we don't need `Weld`, because we still want the joints to move freely?

Reply:
Yes, that's how Weld works. My thought was that, rather than using a separate URDF, we could simply Weld the spring joint for the fixed-springs version. (fixed springs, not no springs)

Reply:
I wonder if welding works in our case. In cassie_v2.urdf, the bodies relation is "thigh - shin" instead of "thigh - knee_spring - shin"

Reply:
I'm not sure why it wouldn't work. Each spring acts on a joint (which connects two bodies). Those two bodies could just be welded.

Reply:
Oh I think that the knee spring is _not_ connected to both thigh and shin in `cassie_v2.urdf`?
I think it's look like: thigh ------ shin
                                       \
                                        -------knee spring
The only joint that `knee_spring` body is connected to is `knee_spring_left_fixed`.
https://github.com/DAIRLab/dairlib/blob/c6f4fc414010c03c9a0801f85adc500a1a737531/examples/Cassie/urdf/cassie_v2.urdf#L733

Reply:
Resolved f2f.